src/serde/mod.rs

```diff
@@ -5,9 +5,41 @@
 mod de;
 mod ser;
 
+use core::fmt;
+
 pub use de::from_ipld;
 pub use ser::{to_ipld, Serializer};
 
+/// Error during Serde operations.
+#[derive(Clone, Debug)]
+#[non_exhaustive]
+pub enum SerdeError {
+    /// Error message describing the error.
+    Message(String),
+}
```

Comment:
If this can only be a string message (it looks like that might be the case), I take back what I said earlier. There's no need to make this an enum in that case (we won't have any other cases, unless I'm mistaken).

My previous point was about preserving information by representing errors as structured data when we have enough information to do so. If we only have a string, we're SOL anyways.

Reply:
Honestly, I don't think deserialize versus serialize is important to distinguish in this case (the user can just annotate the error if they care). And/or, if we _did_ want to distinguish those cases, I'd just have two errors.

But again, I'm just quibbling at this point.

Reply:
Good point, usually you know whether you are serializing or deserializing.